framework/meta/src/cmd/contract/generate_proxy/proxy_gen_main.rs

```diff
@@ -14,6 +17,35 @@ impl MetaConfig {
             write_proxy_with_explicit_path(&proxy_config, self);
         }
     }
+
+    pub fn compare_proxy(&mut self) {
+        for proxy_config in self.sc_config.proxy_configs.clone() {
+            compare_proxy_explicit_path(&proxy_config, self);
+        }
+    }
+}
+
+fn compare_proxy_explicit_path(proxy_config: &ProxyConfigSerde, meta_config: &MetaConfig) { 
+    let err_msg = "Contract has been modified and proxies have not been updated. Regenerate proxies to avoid inconsistencies.".to_string();
+    
+    let contract_abi = extract_contract_abi(proxy_config, meta_config);
+    let mut temp = create_file("temp");
+    let mut proxy_generator =
+    ProxyGenerator::new(meta_config, &mut temp, proxy_config, contract_abi);
+    proxy_generator.write_proxy_to_file();
```

Comment:
You can keep it memory, and compare it there, I think it is a little easier.

The proxy generator needs to be able to output to a String